include/crocoddyl/multibody/costs/contact-cop-position.hpp

```diff
@@ -53,17 +53,20 @@ class CostModelContactCoPPositionTpl : public CostModelAbstractTpl<_Scalar> {
                         const Eigen::Ref<const VectorXs>& u);
   virtual boost::shared_ptr<CostDataAbstract> createData(DataCollectorAbstract* const data);
 
-  const FootGeometry& get_footGeom() const;
+  const std::size_t& get_frame_id() const;
+  const Vector2s& get_cop_region() const;
+  const Vector3s& get_normal() const;
 
+ protected:
   using Base::activation_;
   using Base::nu_;
   using Base::state_;
   using Base::unone_;
 
-  protected: 
-    FootGeometry foot_geom_; //!< frame name and geometrical dimension of the contact foot
-    const Vector3s normal_; //!< vector normal to the contact surface 
-    Vector3s foot_pos_; //!< position of the foot w.r.t. the ground plane
+ private: 
+  const std::size_t frame_id_; //!< contact frame id
```

Comment:
Do use `const` for all private objects. We might want to change its values at runtime